python/dgl/distributed/server.py

```diff
@@ -2,19 +2,85 @@
 
 """Functions used by server."""
 
-def start_server(clients):
-    """Start server.
+import time
 
-    This is a blocking function -- it returns only when the server receives
-    shutdown command from clients.
+from . import rpc
+from .server_state import get_server_state
+
+QUEUE_SIZE = 20*1024*1024*1024
+
+def start_server(server_id, ip_config, num_clients, queue_size=QUEUE_SIZE, net_type='socket'):
```

Comment:
@jermainewang last time we discussed, you convinced me that we don't need to know the number of clients?

Reply:
I guess the `num_clients` is needed during initialization when server#0 needs to assign an ID for each client. Our discussion last time was mainly about the barrier behavior and I remember the conclusion was that full barrier will not be required at the moment.